client/src/partials/vouchers/records.js

```diff
@@ -0,0 +1,196 @@
+        customTreeAggregationFinalizerFn: typeAggregation,
+        groupingShowAggregationMenu: false
+      },
+      { field : 'date', displayName : 'TABLE.COLUMNS.DATE', headerCellFilter: 'translate',
+        cellFilter : 'date:"mediumDate"',
+        filter : { condition : filtering.byDate },
+        customTreeAggregationFinalizerFn: timeAggregation,
+        groupingShowAggregationMenu: false
+      },
+      { field : 'description', displayName : 'TABLE.COLUMNS.DESCRIPTION', headerCellFilter: 'translate',
+        groupingShowAggregationMenu: false
+      },
+      { field : 'amount', displayName : 'TABLE.COLUMNS.AMOUNT', headerCellFilter: 'translate',
+        treeAggregationType: uiGridGroupingConstants.aggregation.SUM,
+          groupingShowAggregationMenu: false
+      },
+      { field : 'user', displayName : 'TABLE.COLUMNS.RESPONSIBLE', headerCellFilter: 'translate',
+        groupingShowAggregationMenu: false
+      },
+      { field : 'action', displayName : '...',
+        cellTemplate: linkReceiptTemplate,
+        enableFiltering: false,
+        enableColumnMenu: false
+      }
+    ];
+
+  // register API
+  vm.gridOptions.onRegisterApi = onRegisterApi;
+
+  // expose function
+  vm.getType = getType;
+  vm.isDefined = isDefined;
+  vm.showReceipt = showReceipt;
+
+  // startup
+  startup();
+
+  // API register function
+  function onRegisterApi(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  // Grid Aggregation
+  function typeAggregation(aggregation) {
+    var type = getType(aggregation.groupVal);
+    aggregation.rendered = type ? $translate.instant(type.text) : $translate.instant('FORM.LABELS.UNDEFINED');
+  }
+
+  // Time Aggregation
+  function timeAggregation(aggregation) {
+    var date = new Date(aggregation.groupVal);
+    var time = date.getHours() + ':' + date.getMinutes();
+    aggregation.rendered = aggregation.groupVal ? date.toDateString().concat('  (', time, ')') : null;
+  }
+
+  // isDefined Type
+  function isDefined(row) {
+    return row.uuid && (row.origin_id === null || row.origin_id === undefined);
+  }
+
+  // get vouchers type
+  function getType(originId) {
+    if (originId === null || originId === undefined) { return {}; }
+
+    return Vouchers.transferType.filter(function (item) {
+      return item.id === originId;
+    })[0];
+  }
+
+  // enable filter
+  function toggleFilter() {
+    vm.filterEnabled = !vm.filterEnabled;
+    vm.bcButtons[0].color = vm.filterEnabled ? 'btn-default active' : 'btn-default';
+    vm.gridOptions.enableFiltering = vm.filterEnabled;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.ALL);
+  }
+
+  // search voucher
+  function search() {
+    Modal.openDateInterval()
+    .then(function (dateInterval) {
+      vm.dateInterval = dateInterval;
+      return Vouchers.read(null, vm.dateInterval);
+    })
+    .then(function (list) {
+      // need only one line of the voucher transaction
```

Comment:
Why?  Can we do this in SQL to avoid shipping back large transactions to the client?